### PR TITLE
Dark mode

### PR DESCRIPTION
**Pull request by jason-fox**
### Light

Screenshot 2022-05-22 at 16 17 07

### Dark
Screenshot 2022-05-22 at 16 16 53

- add   5.2.0-beta1 JS and  deploy-preview CSS
- remove conflicting defaults
- add dark mode toggle
- amend CSS

### CODE REVIEW

**Comment by jason-fox** [comment]
This will remain in draft until Bootstrap 5.3 officially supports dark mode.

File: plugin.xml [hunk 1]
```diff
@@ -77,6 +77,14 @@
     >
       <val>yes</val>
       <val default="true">no</val>
+    </param>
+     <param
```
Comment (infotexture) [inline plugin.xml:81]:
```suggestion
    <param
```

**Comment by infotexture** [comment]
> This will remain in draft until Bootstrap 5.3 officially supports dark mode.

Thanks, sounds like a nice addition and good approach. We can test & refine here, and ship it when the official upstream support lands.

**Comment by jason-fox** [comment]
Bootstrap 5.3-alpha has landed. This can now be reviewed.

**Comment by infotexture** [comment]
@jason-fox I pulled the latest changes here and tried building the DITA-OT docs output. 

I saw the new dark mode toggle in the menu, but it had no effect.

Also tried building the sample document from the plugin itself:

    dita --project=.github/dita-ot/html.xml -o out/sample-document-pr-42-dark-mode-new

In this case, I didn't get the dark mode toggle at all.

Do I need to pass an extra option to the CLI, or am I missing something else? 🤔

**Comment by jason-fox** [comment]
Yes at the moment you need `--dark.mode.include=yes`. This could be altered to be included by default of course. Either way, there will need to be some documentation added

**Comment by jason-fox** [comment]
Documentation added: 4b9c3bb

**Comment by infotexture** [comment]
> Yes at the moment you need `--dark.mode.include=yes`. This could be altered to be included by default of course. 

Got it. Built the DITA-OT docs output with this flag and got the toggle as expected. ✔️ 

I see you [added the parameter to the project file](https://github.com/jason-fox/dita-bootstrap/blob/feature/dark-mode/.github/dita-ot/html.xml#L29) used to generate the sample document, and thought that would be enough to enable the toggle there too, but it wasn't. 🤔 

If I _also_ pass the CLI flag at build time, I get the dark mode rendering in the sample doc, but **not** the toggle in the menu bar.

Is that the way it's supposed to work, or is there still a bug somewhere?

**Comment by infotexture** [comment]
Regarding defaults, I'd be inclined to enable this by default, and set it to `auto`.

That would make the feature more discoverable, as people would see the toggle. 

Anyone that doesn't want it could disable it by setting `--dark.mode.include=no`.

What do you think?

**Comment by infotexture** [comment]
I suggest we merge this soon to `develop`, but maybe wait to mint a new release until Bootstrap 5.3 is officially released.

According to the versioning strategy in https://github.com/infotexture/dita-bootstrap/issues/8#issuecomment-867405771, I'd suggest we call that next plug-in release `5.3.3` 
_(meaning Bootstrap 5.3 for DITA-OT 3)_.

Although DITA-OT 4 has been released, nothing here really requires the new toolkit version, and I'd be inclined to leave it that way for now, so it can also be used with previous toolkit versions.

**Comment by jason-fox** [comment]
> Regarding defaults, I'd be inclined to enable this by default

For consistency, I have switched icons, popovers, tooltips and dark-mode toggler support to include by default. I have also added documentation on how to add toggler HTML to the included menu bar.

>  thought that would be enough 

The header needed updating too - the docs should now explain this.

> wait until Bootstrap 5.3 is officially released.

Yes.

**Review by infotexture — APPROVED** [review]
This all looks good, so I think we're ready to merge.

The toggle icon is static and doesn't reflect the selected mode like in the [Bootstrap Blog](https://blog.getbootstrap.com), but we may need to live with that since the header files are stored in XML and not as easily manipulated via JavaScript as the normal HTML versions.

**Comment by jason-fox** [comment]
> The toggle icon is static and doesn't reflect the selected mode

Fixed  584d47f

**Review by infotexture — APPROVED** [review]
@jason-fox Nice updates, thanks! 🙏

**Pull request merged**